airflow/models/dagcode.py

```diff
@@ -54,11 +56,12 @@ class DagCode(Base):
 
     __tablename__ = "dag_code"
     id = Column(UUIDType(binary=False), primary_key=True, default=uuid6.uuid7)
-    fileloc_hash = Column(BigInteger, nullable=False)
+    dag_id = Column(String(ID_LEN), nullable=False)
```

Comment:
Not leaving this at the file feels like it'd be pretty wasteful for DAG factories, no?

Reply:
I don't fully understand, but I added the dag_id to make it easy to query. Currently, the REST API goes through the dag version before getting the latest dagcode. Do you mean I should have left the fileloc_hash?

Reply:
By DAG makes it a lot simpler when you consider versioning is done by DAG.